### PR TITLE
refactor(mobile): refactor theme management

### PR DESCRIPTION
**Pull request by dvbthien**
This pull request refactors the theme management in the mobile application. The changes include:
- Splitting and refactoring theme-related files: The theme-related code has been reorganized for better maintainability and readability.
- Updating theme providers: Modified the theme providers to handle the color scheme and dynamic theming in Android.

I have tested it on both IOS and Android
How to test it: Change the theme and use dynamic theme in Android. Both should work as expected.

Next stage: 
Hi @alextran1502 

For the next pull request, I plan to declare common colors, font sizes, styles, paddings, and borders to ensure a consistent interface throughout the app. 

Additionally, I've found two font options: Manrope and InterTight. I think they both look good. I'll post images of both fonts for your review. If you prefer one over the other, please let me know. I'll apply the chosen font to both the web and mobile app.

Here's a link to the image of the font: https://imgur.com/a/c5QAib0

What are your thoughts?

### CODE REVIEW

**Comment by alextran1502** [comment]
Thank you for the PR! Yes, we should standardize on a UI for form/card for consistency. 

I haven't found a font that expresses the App UI besides Overpass yet. And I have tried a few. I will return to the font topic another time.

**Comment by dvbthien** [comment]
Thanks for your reply. I will create a pull request for common colors, font sizes, styles, paddings, and borders. I plan to refactor it step by step. Feels happy to contribute to this project.

File: mobile/lib/theme/dynamic_theme.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+import 'package:dynamic_color/dynamic_color.dart';
+
+import 'package:immich_mobile/theme/theme_data.dart';
+
+class DynamicTheme {
```
Comment (shenlong-tanwen) [inline mobile/lib/theme/dynamic_theme.dart:6]: This could very well be an abstract final class as all the members / fields are static

Comment (dvbthien) [reply mobile/lib/theme/dynamic_theme.dart]: Thanks for your reivew. I have made the change.

File: mobile/lib/constants/colors.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  lime,
+  green,
+  cyan,
+  slateGray
+}
+
+const ImmichColorPreset defaultColorPreset = ImmichColorPreset.indigo;
+const String defaultColorPresetName = "indigo";
+
+const Color immichBrandColorLight = Color(0xFF4150AF);
```
[the remark this answers is not in the document]
Comment (dvbthien) [reply mobile/lib/constants/colors.dart:19]: Hi @shenlong-tanwen,

Thanks for your suggestion. Can you explain it to me in more detail? If you have any ideas for unifying colors and themes across the app, let me know. I'll implement them in upcoming PRs. I just started coding in Flutter, so your ideas will help me a lot.

**Comment by dvbthien** [comment]
> A lot of these changes are related to const/linting rule. Can you make all those changes in a separate PR?

Sure, I handled that in the last commit. Thanks for the suggestion!

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**